### PR TITLE
Document how to run Synapse

### PR DESCRIPTION
**Pull request by andybalaam**
I was trying to run Synapse today and couldn't remember how to do it. If I missed this information somewhere else, maybe a link from the contributing guide would be useful?

I'm not 100% sure what I've written here is right, but if it's not I still hope we can add something similar, because I got stuck at this point.

Signed-off-by: Andy Balaam <andy.balaam@matrix.org>

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

File: docs/development/contributing_guide.md [hunk 1]
```diff
@@ -78,6 +78,17 @@ poetry install --extras all
 
 This will install the runtime and developer dependencies for the project.
 
+Create a config file:
```
Comment (DMRobertson) [inline docs/development/contributing_guide.md:81]: Perhaps?

```suggestion
## Running Synapse via poetry

To start a local instance of Synapse in the locked poetry environment, create a config file:
```

**Review by DMRobertson — APPROVED** [review]
Otherwise, many thanks!

File: docs/development/contributing_guide.md [hunk 2]
```diff
@@ -78,6 +78,19 @@ poetry install --extras all
...
+To start a local instance of Synapse in the locked poetry environment, create a config file:
+
+```sh
+cp docs/sample_config.yaml homeserver.yaml
+```
+
+Now edit homeserver.yaml, and run Synapse with:
+
+```sh
+poetry run python synapse/app/homeserver.py -c homeserver.yaml
```
Comment (clokep) [inline docs/development/contributing_guide.md:92]: For consistency with how we do this everywhere else I'd suggest:

```suggestion
poetry run python -m synapse.app.homeserver -c homeserver.yaml
```

Comment (clokep) [reply docs/development/contributing_guide.md:92]: (This also helps with path issues if you're in a different directory.)

**Pull request merged**